docs/design/values.md

```diff
@@ -0,0 +1,920 @@
+  // modify them or take their address.
+  return x + y;
+}
+```
+
+Value bindings require the matched expression to be a _value expression_,
+converting it into one as necessary.
+
+A _variable pattern_ can be introduced with the `var` keyword to create an
+object with storage when match.d Every binding pattern name introduced within a
+variable pattern is called a _variable binding_ and forms a
+[_reference expression_](#reference-expressions) to an object within the
+variable pattern's storage when used. Variable patterns require their matched
+expression to be an _initializing expression_ and provide their storage to it to
+be initialized.
+
+```carbon
+fn MutateThing(ptr: i64*);
+
+fn Example() {
+  // Both `1` and `2` here start as value expressions. That is a match for `1`.
+  // For `2`, the variable binding requires it to be converted to an
+  // initializing expression by using the value `2` to initialize the provided
+  // variable storage that `y` will refer to.
+  let (x: i64, var y: i64) = (1, 2);
+
+  // Allowed to take the address and mutate `y` as it is a durable reference
+  // expression.
+  MutateThing(&y);
+
+  // ❌ This would be an error though due to trying to take the address of the
+  // value expression `x`.
+  MutateThing(&x);
+}
+```
+
+### Local variables
+
+A local binding pattern can be introduced with either the `let` or `var`
+keyword. The `let` introducer begins a value pattern just like the default
+patterns in other contexts. The `var` introducer works exactly the same as
+introducing the pattern in some other context with `var` -- there's just no need
+for the outer `let`.
+
+-   `let` _identifier_`:` _< expression |_ `auto` _>_ `=` _value_`;`
+-   `var` _identifier_`:` _< expression |_ `auto` _> [_ `=` _value ]_`;`
+
+These are just simple examples of binding patterns used directly in local
+declarations. Local `let` and `var` declarations build on Carbon's general
+[pattern matching](/docs/design/pattern_matching.md) design, with `var`
+declarations implicitly starting off within a `var` pattern while `let`
+declarations introduce patterns that work the same as function parameters and
+others with bindings that produce values by default.
+
+### Consuming function parameters
+
+Just as part of a `let` binding can use a `var` prefix to become a variable
+pattern and bind names that will form reference expressions to the variable's
+storage, so can function parameters:
+
+```carbon
+fn Consume(var x: SomeData) {
+  // We can mutate and use variable that `x` refers to here.
+}
+```
+
+This allows us to model an important special case of function inputs -- those
+that are _consumed_ by the function, either through local processing or being
+moved into some persistent storage. Marking these in the pattern and thus
+signature of the function changes the expression category of arguments in the
+caller, causing them to become _initializing expressions_ that directly
```

Comment:
Become, or be converted to?

Reply:
This was trying to contrast with patterns that *don't* get marked this way, but that is clearly a bit ambiguous.

I've switched the wording around to talk first about a contextual requirement, and then mention how that may in turn trigger a conversion.